ssr/render.js

```diff
@@ -43,8 +45,13 @@ export default function render(renderApp, doc) {
     <script id="documentdata" type="application/json">${escapeDocumentJson}</script>
     `.trim();
     $("#root").after(documentDataTag);
+  } else {
+    // TODO: The day we have SPAs in Yari, we need to pass those details
+    // down to here too.
   }
 
+  $('link[rel="canonical"]').attr("href", canonicalURL);
```

Comment:
I'd rather not because we don't have any other URLs other than the home page and even that I don't want to slip through. The `$('link[rel="canonical"]').attr("href", ...` line should ALWAYS happen. 
At the moment, it's a bit weird because we don't build anything that isn't a doc.

But in a not-so-distant future we will SSR build other pages. That day, I want that line to not have to change but something above needs to change. 

One interesting thought might be that we change the whole signature of the `render` function to look like this:

```diff
-export default function render(renderApp, doc) {
+export default function render(renderApp, doc, url, title, description='') {
```
and call it with `render(app, doc, doc.mdn_url, doc.title, doc.summary)`

Reply:
What I find strange about the current state is that adding a new page without suppliying a doc would it would by default get a canonical URL pointing towards root. How about changing it to:

```
let canonicalPath = null;
if (doc) {
  // ...
  canonicalPath = doc.mdn_url;
}

if (canonicalPath) {
  $('link[rel="canonical"]').attr("href", `https://developer.mozilla.org${canonicalPath}`);
}
```

Reply:
I'm not sure that's an improvement. Yes, it's a bit weird that it points to the root, but it's also no harm and it's a nice safeguard that the URL wouldn't be picked up the Googlebot. I.e. https://pr1234.yari.developer.mozilla.org/en-US/_flaws or something.

Reply:
Fair! Then consider my concerns lifted